sktime/forecasting/moirai_forecaster.py

```diff
@@ -0,0 +1,461 @@
+        context_length=200,
+        patch_size=32,
+        num_samples=100,
+        num_feat_dynamic_real=0,
+        num_past_feat_dynamic_real=0,
+        map_location=None,
+        broadcasting=False,
+        deterministic=False,
+        batch_size=32,
+        use_source_package=False,
+    ):
+        super().__init__()
+        self.checkpoint_path = checkpoint_path
+        self.context_length = context_length
+        self.patch_size = patch_size
+        self.num_samples = num_samples
+        self.num_feat_dynamic_real = num_feat_dynamic_real
+        self.num_past_feat_dynamic_real = num_past_feat_dynamic_real
+        self.map_location = map_location
+        self.broadcasting = broadcasting
+        self.deterministic = deterministic
+        self.batch_size = batch_size
+        self.use_source_package = use_source_package
+
+        if self.broadcasting:
+            self.set_tags(
+                **{
+                    "y_inner_mtype": "pd.DataFrame",
+                    "X_inner_mtype": "pd.DataFrame",
+                    "capability:global_forecasting": False,
+                }
+            )
+
+    # Apply a patch for redirecting imports to sktime.libs.uni2ts
+    @patch.dict("sys.modules", {"uni2ts": sktime.libs.uni2ts})
+    def _instantiate_patched_model(self, model_kwargs):
+        """Instantiate the model from the vendor package."""
+        from sktime.libs.uni2ts.forecast import MoiraiForecast
+
+        self.model = MoiraiForecast.load_from_checkpoint(**model_kwargs)
+        self.model.to(self.map_location)
+
+    def _fit(self, y, X, fh):
+        if fh is not None:
+            prediction_length = max(fh.to_relative(self.cutoff))
+        else:
+            prediction_length = 1
+
+        if self.num_feat_dynamic_real is None and X is not None:
+            self.num_feat_dynamic_real = X.shape[1]
+        if self.num_past_feat_dynamic_real is None:
+            self.num_past_feat_dynamic_real = 0
+
+        model_kwargs = {
+            "prediction_length": prediction_length,
+            "context_length": self.context_length,
+            "patch_size": self.patch_size,
+            "num_samples": self.num_samples,
+            "target_dim": 2,
+            "feat_dynamic_real_dim": self.num_feat_dynamic_real,
+            "past_feat_dynamic_real_dim": self.num_past_feat_dynamic_real,
+        }
+
+        model_kwargs["checkpoint_path"] = hf_hub_download(
+            repo_id=self.checkpoint_path, filename="model.ckpt"
+        )
+
+        # Load model from source package
+        if self.use_source_package:
+            if _check_soft_dependencies("uni2ts", severity="none"):
+                from uni2ts.model.moirai import MoiraiForecast
+
+                self.model = MoiraiForecast.load_from_checkpoint(**model_kwargs)
+                self.model.to(self.map_location)
+        # Load model from sktime
+        else:
+            self._instantiate_patched_model(model_kwargs)
```

Comment:
See comments on the `_instantiate_patched_model` above. Then here you could directly import from our vendored lib.